src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php

```diff
@@ -426,7 +520,27 @@ protected function addSyncPublishedMessageMethod()
     {
         $params = $this->getParams();
         $resource = $this->getParameter('resource')['value'];
+        $parameters = $this->getParameters();
 
+        $sendMappingStatement = '';
+        if (isset($parameters['mapping_resource'])) {
```

Comment:
I think the method body is big now, can you please extract this to a separated method